Classes/Format/Mods.php

```diff
@@ -34,98 +50,206 @@ class Mods implements \Kitodo\Dlf\Common\MetadataInterface
+
+        $this->xml->registerXPathNamespace('mods', 'http://www.loc.gov/mods/v3');
+
+        $this->getAuthors();
+        $this->getHolders();
+        $this->getPlaces();
+        $this->getYears();
+
+        $metadata = $this->metadata;
+    }
+
+    /**
+     * Get "author" and "author_sorting".
+     *
+     * @access private
+     *
+     * @return void
+     */
+    private function getAuthors() {
+        $authors = $this->xml->xpath('./mods:name[./mods:role/mods:roleTerm[@type="code" and @authority="marcrelator"]="aut"]');
+
+        // Get "author" and "author_sorting" again if that was too sophisticated.
         if (empty($authors)) {
             // Get all names which do not have any role term assigned and assume these are authors.
-            $authors = $xml->xpath('./mods:name[not(./mods:role)]');
+            $authors = $this->xml->xpath('./mods:name[not(./mods:role)]');
         }
         if (!empty($authors)) {
             for ($i = 0, $j = count($authors); $i < $j; $i++) {
                 $authors[$i]->registerXPathNamespace('mods', 'http://www.loc.gov/mods/v3');
-                // Check if there is a display form.
-                if (($displayForm = $authors[$i]->xpath('./mods:displayForm'))) {
-                    $metadata['author'][$i] = (string) $displayForm[0];
-                } elseif (($nameParts = $authors[$i]->xpath('./mods:namePart'))) {
-                    $name = [];
-                    $k = 4;
-                    foreach ($nameParts as $namePart) {
-                        if (
-                            isset($namePart['type'])
-                            && (string) $namePart['type'] == 'family'
-                        ) {
-                            $name[0] = (string) $namePart;
-                        } elseif (
-                            isset($namePart['type'])
-                            && (string) $namePart['type'] == 'given'
-                        ) {
-                            $name[1] = (string) $namePart;
-                        } elseif (
-                            isset($namePart['type'])
-                            && (string) $namePart['type'] == 'termsOfAddress'
-                        ) {
-                            $name[2] = (string) $namePart;
-                        } elseif (
-                            isset($namePart['type'])
-                            && (string) $namePart['type'] == 'date'
-                        ) {
-                            $name[3] = (string) $namePart;
-                        } else {
-                            $name[$k] = (string) $namePart;
+
+                $identifier = $authors[$i]->xpath('./mods:nameIdentifier');
+                if ($identifier[0]['type'] == 'orcid' && !empty((string) $identifier[0])) {
+                    $orcidIdParts = explode('/', (string) $identifier[0]);
+                    $orcidId = trim(end($orcidIdParts));
+                    if (!str_contains($orcidId, 'orcid=')) {
```

Comment:
This is only necessary because of a faulty metadata mapping. The ORCID ID should be mapped to `./mods:name/mods:nameIdentifier[@type="orcid"]` as a plain ID value, not as full URL. So there shouldn't be any need to do string magic here in order to get the ID...
This should be fixed in the metadata mapping and removed from the code here.

Reply:
Currently we are getting this data as URL:
```
<author_orcid>https://orcid.org/0000-0001-8356-8483</author_orcid>`
<holder_viaf>http://viaf.org/viaf/170713989</holder_viaf>
```

Should be string magic made in XSLT transformation?

Reply:
Since `mods:nameIdentifier` unfortunately doesn't allow the `@valueURI` attribute (where we could put the whole URL into), I think we need to do the string magic in XSLT in order to get the plain ID.